src/main/java/net/dries007/tfc/api/recipes/BarrelRecipeCoolItems.java

```diff
@@ -0,0 +1,66 @@
+package net.dries007.tfc.api.recipes;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import net.minecraft.init.SoundEvents;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.FluidStack;
+
+import net.dries007.tfc.api.capability.heat.CapabilityItemHeat;
+import net.dries007.tfc.api.capability.heat.IItemHeat;
+import net.dries007.tfc.objects.inventory.ingredient.IIngredient;
+
+import static net.dries007.tfc.objects.fluids.FluidsTFC.FRESH_WATER;
+
+public class BarrelRecipeCoolItems extends BarrelRecipe
+{
+    public BarrelRecipeCoolItems()
+    {
+        super(IIngredient.of((FluidStack) null), IIngredient.of((ItemStack) null), null, (ItemStack) null, 0);
+    }
+
+    @Override
+    public boolean isValidInput(FluidStack inputFluid, ItemStack inputStack)
+    {
+        return IIngredient.of(FRESH_WATER, 1).testIgnoreCount(inputFluid) && inputStack.hasCapability(CapabilityItemHeat.ITEM_HEAT_CAPABILITY, null) && inputStack.getCapability(CapabilityItemHeat.ITEM_HEAT_CAPABILITY, null).getTemperature() > 0;
+    }
+
+    @Override
+    @Nonnull
+    public ItemStack getOutputItem(FluidStack inputFluid, ItemStack inputStack)
+    {
+        IItemHeat heat = inputStack.getCapability(CapabilityItemHeat.ITEM_HEAT_CAPABILITY, null);
+        heat.setTemperature(heat.getTemperature() - 50);
+        return inputStack;
+    }
+
+    @Override
+    @Nullable
+    public FluidStack getOutputFluid(FluidStack inputFluid, ItemStack inputStack)
+    {
+        int multiplier = inputStack.getCount();
+        int retainAmount = inputFluid.amount - multiplier;
+        if (retainAmount > 0)
+        {
+            return new FluidStack(inputFluid.getFluid(), retainAmount);
+        }
+        return null;
+    }
+
+    @Override
+    public boolean shouldRepeat()
```

Comment:
I think this can be removed, as there is no use case for this ever returning false.

Reply:
I think it's essential, it returns false in eveery other recipe, but only returns true in this one, so it can repeat, but it may be used in later recipes